src/Workspaces/Remote/ServiceHub/Services/DiagnosticAnalyzer/DiagnosticComputer.cs

```diff
@@ -37,6 +37,38 @@ internal class DiagnosticComputer
         /// pressure when there are large number of open documents across different projects to be analyzed by background analysis.
         /// </summary>
         private static CompilationWithAnalyzersCacheEntry? s_compilationWithAnalyzersCache = null;
+
+        /// <summary>
+        /// List of high priority diagnostic computation tasks which are currently executing.
+        /// Any new high priority diagnostic request is added to this list before the core diagnostics
+        /// compute call is performed, and removed from this list after the computation finishes.
+        /// Any new normal priority diagnostic request first waits for all the high priority tasks in this list
+        /// to complete, and moves ahead only after this list becomes empty.
+        /// </summary>
+        /// <remarks>
+        /// Read/write access to the list is guarded by <see cref="s_gate"/>.
+        /// </remarks>
+        private static readonly List<Task> s_highPriorityComputeTasks = new();
```

Comment:
Note to self.  Can these be non-static fields?

Reply:
> Note to self. Can these be non-static fields?

I am not sure that is possible as we instantiate a new `DiagnosticComputer` for every diagnostic request.

Reply:
that's fine.  An alternatite would be this being an instance field on the *creator* of hte DiagnosticComputer.  But i don't think you need to change this around much.  This is about perf, and having static info of "something expensvie is going on somewhere" is a reasonable strategy (i think :D).